src/python/pants/backend/jvm/tasks/junit_run.py

```diff
@@ -513,11 +409,29 @@ def _sort_properties(properties_with_tests):
       for i in range(0, len(sorted_tests), stride):
         yield properties, sorted_tests[i:i + stride]
 
+  def _parse(self, test_spec_str):
+    """Parses a test specification string into an object that can yield corresponding tests.
+
+    Tests can be specified in one of four forms:
+
+    * [classname]
```

Comment:
Just two forms:
* [fully qualified classname
* [fully qualified classname]#[methodname]

In order to support [classname] variants of this we'd have to support some sort of policy for finding unqualified classnames which I'm pretty sure we'd all agree we don't want to wade into... but you did below! I vote to stick to the PR purpose and just trim the option help / code to do what it claims and not add new functionality (guessing unqualifed class names). That is better added in a follow up iff there is agreement this is a wise feature to add.